File: pingyougou_cart_interface/src/main/java/com/pinyougou/cart/service/CartService.java
package com.pinyougou.cart.service;

import com.pinyougou.pojogroup.Cart;

import java.util.List;

/**
 * 购物车服务接口
 */
public interface CartService {

    /**
     * 添加商品到购物车
     * @param cartList
     * @param itemId
     * @param num
     * @return
     */
    public List<Cart> addGoodsToCartList(List<Cart> cartList, Long itemId, Integer num);


    /**
     * 从redis中查询购物车
     * 大key cartList  小key 用户名传递过去
     * @return
     */
    public List<Cart> findCartListFromRedis(String username);

    /**
     * 将购物车保存到 redis
     * @param username
     * @param cartList
     */
    public void saveCartListToRedis(String username,List<Cart> cartList);


    /**
     * 合并购物车
     * @param cartList1
     * @param cartList2
     * @return
     */
    public List<Cart> mergeCartList(List<Cart> cartList1, List<Cart> cartList2);


    /**
     * 保存需要支付购物车的订单到redis中
     * @param cartList
     */
    public void selectOrderItemListPayToRedis(String username,List<Cart> cartList);


    /**
     * 查询需要支付的订单购物车列表返回
     * @return
     */
    public List<Cart> findOrderItemListPayFromRedis(String username);

}

File: pingyougou_cart_web/src/main/webapp/js/service/cartService.js
app.service("cartService",function ($http) {
    //查询购物车列表
    this.findCartList=function () {
        return $http.get("cart/findCartList.do");
    }

    //添加商品到购物车
    this.addGoodsToCartList=function (itemId,num) {
        return $http.get("cart/addGoodsToCartList.do?itemId="+itemId+"&num="+num);
    }


    //求合计
    this.num=function (cartList) {
        var totalValue = {totalNum: 0, totalMoney: 0.00};//合计实体
        
        for(var i = 0; i < cartList.length; i++) {
            var cart = cartList[i];//购物车对象
            for(var j = 0; j < cart.orderItemList.length; j++) {
                var orderItem = cart.orderItemList[j];//购物车明细
                totalValue.totalNum+=orderItem.num;//累加数量
                totalValue.totalMoney+=orderItem.totalFee;//累加金额
            }
        }

        return totalValue;
        
    };
    
    
    //获取地址列表
    this.findAddressList=function () {
        return $http.get('address/findListByLoginUser.do');
    }

    //保存订单
    this.submitOrder=function (order) {
        return $http.post("order/add.do", order);
    }


    //增加
    this.add=function(entity){
        return  $http.post('address/add.do',entity );
    }

    //修改
    this.update=function(entity){
        return  $http.post('address/update.do',entity );
    }

    //异步保存到redis中 选中的订单结算
    this.selectOrderItemListPay=function (selectOrderItemList) {
        return $http.post('cart/selectOrderItemListPayToRedis.do',selectOrderItemList);
    }

    //查询出来需要支付的结算订单
    this.findOrderItemListPay=function () {
       return $http.get('findOrderItemListPayFromRedis.do');
    }
    

});